servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/MergeCompletable.java

```diff
@@ -15,77 +15,54 @@
  */
 package io.servicetalk.concurrent.api;
 
-import javax.annotation.Nullable;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
 
 import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.atomic.AtomicIntegerFieldUpdater.newUpdater;
 
-/**
- * A {@link Completable} implementation for merging {@link Completable}s.
- */
 final class MergeCompletable extends AbstractMergeCompletableOperator {
-
-    @Nullable
     private final Completable[] others;
-    @Nullable
-    private final Completable onlyOther;
     private final boolean delayError;
 
-    /**
-     * New instance.
-     * @param delayError {@code true} to wait until all {@code others} complete before propagating an error.
-     *                   {@code false} to fail fast and propagate an error on the first
-     *                   {@link Subscriber#onError(Throwable)} observed.
-     * @param original {@link Completable} to merge with {@code others}.
-     * @param others {@link Completable}s to merge with {@code original}.
-     */
-    MergeCompletable(boolean delayError, Completable original, Executor executor, Completable... others) {
+    private MergeCompletable(boolean delayError, Completable original, Executor executor, Completable... others) {
         super(original, executor);
         this.delayError = delayError;
-        switch (others.length) {
-            case 0:
-                throw new IllegalArgumentException("At least one Completable required to merge");
```

Comment:
Consider reverting this check until we fix API as described above.

Reply:
I don't think we need to prevent the empty use case you described. There is nothing incorrect about passing an empty array or empty iterator, and I think we would want consistency between the two. If folks are passing empty collections they will just incur some overhead. If we have an API which prevents it that is fine, but until then lets just allow it consistently.